tests/migration/namespace.go

```diff
@@ -0,0 +1,391 @@
+			return nil
+		}
+		// Verify migration is gone
+		Eventually(func() *virtv1.VirtualMachineInstanceMigration {
+			migration, err := virtClient.VirtualMachineInstanceMigration(migration.Namespace).Get(context.Background(), migration.Name, metav1.GetOptions{})
+			if k8serrors.IsNotFound(err) {
+				return nil
+			}
+			return migration
+		}, 30*time.Second, 1*time.Second).Should(BeNil())
+		return nil
+	}
+
+	deleteVM := func(vm *v1.VirtualMachine) {
+		err := virtClient.VirtualMachine(vm.Namespace).Delete(context.Background(), vm.Name, metav1.DeleteOptions{})
+		if k8serrors.IsNotFound(err) {
+			return
+		}
+		Expect(err).ToNot(HaveOccurred())
+		// Verify VM is gone
+		Eventually(func() *virtv1.VirtualMachine {
+			vm, err := virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			if k8serrors.IsNotFound(err) {
+				return nil
+			}
+			return vm
+		}, 30*time.Second, 1*time.Second).Should(BeNil())
+	}
+
+	deleteDV := func(dv *cdiv1.DataVolume) {
+		err := virtClient.CdiClient().CdiV1beta1().DataVolumes(dv.Namespace).Delete(context.Background(), dv.Name, metav1.DeleteOptions{})
+		if k8serrors.IsNotFound(err) {
+			return
+		}
+		Expect(err).ToNot(HaveOccurred())
+		// Verify DV is gone
+		Eventually(func() *cdiv1.DataVolume {
+			dv, err := virtClient.CdiClient().CdiV1beta1().DataVolumes(dv.Namespace).Get(context.Background(), dv.Name, metav1.GetOptions{})
+			if k8serrors.IsNotFound(err) {
+				return nil
+			}
+			return dv
+		}, 30*time.Second, 1*time.Second).Should(BeNil())
+	}
+
+	updateRunStrategy := func(vm *virtv1.VirtualMachine, strategy *virtv1.VirtualMachineRunStrategy) {
+		Eventually(func() error {
+			vm.Spec.RunStrategy = strategy
+			_, err = virtClient.VirtualMachine(vm.Namespace).Update(context.Background(), vm, metav1.UpdateOptions{})
+			if err != nil {
+				// Ignore the error from the get.
+				vm, _ = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			}
+			return err
+		}, 60*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+	}
+
+	Context("container disk", func() {
+		var (
+			sourceVMI, targetVMI *virtv1.VirtualMachineInstance
+			sourceVM, targetVM   *virtv1.VirtualMachine
+		)
+
+		It("should live migration a container disk vm, several times", func() {
+			sourceVMI = libvmifact.NewCirros(
+				libvmi.WithNamespace(testsuite.NamespaceTestDefault),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			)
+			targetVMI = sourceVMI.DeepCopy()
+			targetVMI.Namespace = testsuite.NamespaceTestAlternative
+			sourceVM = createAndStartVMFromVMISpec(sourceVMI)
+			num := 4
+			for i := 0; i < num; i++ {
+				var sourceMigration, targetMigration *virtv1.VirtualMachineInstanceMigration
+				var expectedVMI *virtv1.VirtualMachineInstance
+				sourceRunStrategy := sourceVM.Spec.RunStrategy
+				// execute a migration, wait for finalized state
```

Comment:
Consider using calls to `By()` instead of comments. Also in loops, it's useful for debugging to include the iteration number in some `By()` call to know how far we went.

Reply:
Sure